canonical-web-and-design/maas-ui
# feat(ui): add additional device states

## Done

- Update the device slice to add active, eventErrors, selected and statuses.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Visit /r/dashboard (click on the header logo).
- Open the menu for one of the table rows and click 'Add discovery...'
- In 'Type' field select 'Interface'.
- Choose a device and click 'Save'.
- The field should close and you should get a message "An interface has been added."
- Open the redux dev tools and look for "device/createInterfaceStart" which should set the `creatingInterface` status for the device you selected.

## Fixes

Fixes: canonical-web-and-design/app-tribe#518.
